app/signup/page.tsx

```diff
@@ -0,0 +1,135 @@
+"use client";
+
+import { useState } from "react";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { Card, CardHeader, CardContent, CardFooter, CardTitle, CardDescription } from "@/components/ui/card";
+import { Progress } from "@/components/ui/progress";
+import { Upload } from "lucide-react";
+import Image from "next/image";
+import Link from "next/link";
+
+const FormInput = ({ label, type, placeholder }) => (
+  <div>
+    <label className="block text-sm font-medium text-gray-700">{label}</label>
+    <Input type={type} placeholder={placeholder} className="w-full mt-1 rounded-full" />
+  </div>
+);
+
+export default function SignupPage() {
+  const [page, setPage] = useState(1);
+  const [profilePhoto, setProfilePhoto] = useState(null);
+
+  const handleProfileUpload = (e) => {
+    const file = e.target.files[0];
+    if (file) {
+      setProfilePhoto(URL.createObjectURL(file));
+    }
+  };
+
+  const renderPageContent = () => {
```

Comment:
cleaner approach would be to convert each component you return here to its own component, like maybe `FormStepOne` and `FormStepTwo`

this is related to a comment below